src/OpenTK.Mathematics/Matrix/Matrix2.cs

```diff
@@ -560,10 +560,15 @@ public static void Invert(in Matrix2 mat, out Matrix2 result)
 
             var invDet = 1f / det;
 
+            // Because the c# jit assumes noalias for byref types we need to
+            // save this value as the write to result.Row0.X could change the
+            // value of mat.Row0.X.
```

Comment:
This doesn't sound very correct? The CLI spec and JIT most definitely *do* believe these to be aliasing, which is the whole core of the problem.

Also this might not actually hurt performance because loading into a var like this shouldn't have any cost, and it removes the second duplicate load (that only comes in under undesired aliasing problems). Did you profile this/check ASM?

Reply:
These changes are extensively benchmarked (as you can see above, the disassembly is also provided) and I've done a deep dive into the JIT and CLR.
The JIT and CLR assumes no aliasing for byref values, I've checked dotnet/runtime issues and looked at disassembly output to verify this.

Reply:
After talking on the discord we came to the conclusion that I was using the wrong terminology and that it's most definitely assuming `alias` instead of `noalias`. This is fixed now.